stdlib/Distributed/test/distributed_exec.jl

```diff
@@ -1530,6 +1546,64 @@ cluster_cookie("foobar") # custom cookie
 npids = addprocs_with_testenv(WorkerArgTester(`--worker=foobar`, false))
 @test remotecall_fetch(myid, npids[1]) == npids[1]
 
+VERBOSE && println("\nTesting connect timeout...")
+# tests for connect timeout to worker
+struct ConnectTimeoutTester <: ClusterManager
+    block::Channel
+    function ConnectTimeoutTester()
+        new(Channel(1))
+    end
+end
+
+function launch(manager::ConnectTimeoutTester, params::Dict, launched::Array, c::Condition)
+    dir = params[:dir]
+    exename = params[:exename]
+    exeflags = params[:exeflags]
+
+    cmd = `$exename $exeflags --bind-to $(Distributed.LPROC.bind_addr) --worker=`
+    cmd = pipeline(detach(setenv(cmd, dir=dir)))
+    io = open(cmd, "r+")
+
+    wconfig = WorkerConfig()
+    wconfig.process = io
+    wconfig.io = io.out
+    push!(launched, wconfig)
+
+    notify(c)
+end
+manage(::ConnectTimeoutTester, ::Integer, ::WorkerConfig, ::Symbol) = nothing
+function connect(manager::ConnectTimeoutTester, pid::Int, config::WorkerConfig)
+    # block for 360 seconds
+    Timer(360) do timer
+        put!(manager.block, nothing)
+    end
+    (bind_addr, port) = Distributed.read_worker_host_port(config.io)
+    wait(manager.block)
+    (s, bind_addr) = Distributed.connect_to_worker(bind_addr, port)
+    config.bind_addr = bind_addr
+    config.connect_at = (bind_addr, port)
+    if config.io !== nothing
+        let pid = pid
+            redirect_worker_output(pid, Base.notnothing(config.io))
+        end
+    end
+    (s, s)
+end
+
+nprocs()>1 && rmprocs(workers())
+cluster_cookie("")
+npids, t, bytes, gctime, memallocs = @timed try
+    withenv("JULIA_WORKER_TIMEOUT"=>"5") do
+        addprocs_with_testenv(ConnectTimeoutTester())
+    end
+catch ex
+    []
+end
+@test length(npids) == 0
+@test nprocs() == 1
+@test abs(t - 10.0) <= 5.0
```

Comment:
```suggestion
@test abs(t - 10.0) <= 25.0
```

Reply:
I suppose you suggested this because you suspect the allowed timeout is too tight and we sometimes may exceed it?